src/viam/api/CMakeLists.txt

```diff
@@ -0,0 +1,291 @@
+
+# TODO: This is analogized from
+# https://gitlab.kitware.com/cmake/cmake/-/tree/master/Help/guide/importing-exporting/MathFunctionsComponents
+# but it doesn't really seem to do anything. Everything works fine if
+# it is removed, and naming the alias something other than `viamcpp`
+# doesn't make it available under that alias anyway, as one might
+# hope. What is it supposed to do?
+add_library(viam-cpp-sdk::viamapi ALIAS viamapi)
+
+target_sources(viamapi
+  PRIVATE
+    ${PROTO_GEN_DIR}/app/v1/app.grpc.pb.cc
+    ${PROTO_GEN_DIR}/app/v1/app.pb.cc
+    ${PROTO_GEN_DIR}/app/v1/robot.grpc.pb.cc
+    ${PROTO_GEN_DIR}/app/v1/robot.pb.cc
+    ${PROTO_GEN_DIR}/common/v1/common.grpc.pb.cc
+    ${PROTO_GEN_DIR}/common/v1/common.pb.cc
+    ${PROTO_GEN_DIR}/component/camera/v1/camera.grpc.pb.cc
+    ${PROTO_GEN_DIR}/component/camera/v1/camera.pb.cc
+    ${PROTO_GEN_DIR}/component/generic/v1/generic.grpc.pb.cc
+    ${PROTO_GEN_DIR}/component/generic/v1/generic.pb.cc
+    ${PROTO_GEN_DIR}/component/motor/v1/motor.grpc.pb.cc
+    ${PROTO_GEN_DIR}/component/motor/v1/motor.pb.cc
+    ${PROTO_GEN_DIR}/google/api/annotations.pb.cc
+    ${PROTO_GEN_DIR}/google/api/http.pb.cc
+    ${PROTO_GEN_DIR}/google/api/httpbody.pb.cc
+    ${PROTO_GEN_DIR}/module/v1/module.grpc.pb.cc
+    ${PROTO_GEN_DIR}/module/v1/module.pb.cc
+    ${PROTO_GEN_DIR}/robot/v1/robot.grpc.pb.cc
+    ${PROTO_GEN_DIR}/robot/v1/robot.pb.cc
+    ${PROTO_GEN_DIR}/tagger/v1/tagger.grpc.pb.cc
+    ${PROTO_GEN_DIR}/tagger/v1/tagger.pb.cc
+  PUBLIC FILE_SET viamapi_includes TYPE HEADERS
+    BASE_DIRS
+      ${PROTO_GEN_DIR}/../..
+    FILES
+      ${PROTO_GEN_DIR}/../../viam/api/app/v1/app.grpc.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/app/v1/app.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/app/v1/robot.grpc.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/app/v1/robot.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/common/v1/common.grpc.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/common/v1/common.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/component/camera/v1/camera.grpc.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/component/camera/v1/camera.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/component/generic/v1/generic.grpc.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/component/generic/v1/generic.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/component/motor/v1/motor.grpc.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/component/motor/v1/motor.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/google/api/annotations.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/google/api/http.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/google/api/httpbody.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/module/v1/module.grpc.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/module/v1/module.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/robot/v1/robot.grpc.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/robot/v1/robot.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/tagger/v1/tagger.grpc.pb.h
+      ${PROTO_GEN_DIR}/../../viam/api/tagger/v1/tagger.pb.h
+)
```

Comment:
There is a meaningful change here. I hadn't fully understood that having a `BASE_DIRS` set on a `FILE_SET` affected the include paths associated with the target. But if does! The way we had it before (which gh declines to show us), resulted in the addition of a useless include path. While what is here is uglier, it is more correct because it creates a file set that "knows" that it should be installed under `CMAKE_INCLUDE_DIR/viam/api`.

Reply:
Yes, but that was a risk from before too. It would be better if this file set (and, really, all the other lists in this file instead somehow captured all the outputs from the `buf` invocation. There is a comment to that effect elsewhere in this file. Some sort of globbing is probably required. But for now we are manually curating this list, and the set of "supported" protos that will be in the API list will a subset (possibly improper) of the API.